traitlets/traitlets.py

```diff
@@ -1644,6 +1644,21 @@ def trait_events(cls, name=None):
                         events[k] = v
         return events
 
+    def clear(self, *names, **metadata):
```

Comment:
`clear` is much too generic a name to put on a base class like HasTraits. The namespace of methods is *extremely* precious. All trait-related methods should contain `trait`, and ideally start with `trait_`. Maybe `.trait_clear()`?